vsintegration/src/FSharp.Editor/Completion/HashDirectiveCompletionProvider.fs

```diff
@@ -82,9 +82,18 @@ type internal HashDirectiveCompletionProvider(workspace: Workspace, projectInfoM
            )
         |> Seq.toList
 
+    static member Create(workspace: Workspace, projectInfoManager: FSharpProjectOptionsManager) =
+        let completions =
+            [
+                HashCompletion.Create("""\s*#load\s+(@?"*(?<literal>"[^"]*"?))""", [ ".fs"; ".fsx" ], useIncludeDirectives = true)
+                HashCompletion.Create("""\s*#r\s+(@?"*(?<literal>"[^"]*"?))""", [ ".dll"; ".exe" ], useIncludeDirectives = true)
+                HashCompletion.Create("""\s*#I\s+(@?"*(?<literal>"[^"]*"?))""", [ "\x00" ], useIncludeDirectives = false)
+            ]
```

Comment:
have you considered:
#if, #else, #endif, and #time.

Reply:
All four could be interesting. Since this mainly just calls into Roslyn's file system completion it'd have to be handled a little differently though.

Reply:
For the first three, we'd need to look at the environment and grab MSBuild defines. `#time` would be just a static `"on"`/`"off"`.